File: 01.FactoryMethod/java/src/main.java
import java.lang.System;

public class main {

	public static void main(String[] args){
		
		System.out.println("");
		System.out.println("Get factory");
		shapeFactory Shapefactory = new shapeFactory();
		
		System.out.println("");
		System.out.println("Get Circle");
		Shape circle = Shapefactory.getShape("CIRCLE");
		circle.draw();
		
		System.out.println("");
		System.out.println("Get Rectangle");
		Shape rectangle = Shapefactory.getShape("RECTANGLE");
		rectangle.draw();
		
		System.out.println("");
		System.out.println("Get Square");
		Shape square = Shapefactory.getShape("SQUARE");
		square.draw();
		
		
	}
}

File: 01.FactoryMethod/cpp/src/main.cpp
#include <cstdlib>
#include <iostream>
#include "./factory/shapeFactory.h"
#include "./shape/shape.h"

int main(int argc, char** argv) {
    
    shapeFactory* ShapeFactory = new shapeFactory();  

    shape* circle = ShapeFactory->getShape("circle");
    circle->draw();
    
    shape* square = ShapeFactory->getShape("square");
    square->draw();
    
    shape* rectangle = ShapeFactory->getShape("rectangle");
    rectangle->draw();
    
    std::cout << "This is factory method" << std::endl;
    std::cout.flush();
    
    return 0;
}

